tests_metricflow/snapshots/test_conversion_metric_rendering.py/SqlQueryPlan/BigQuery/test_conversion_metric_with_time_constraint__plan0_optimized.sql

```diff
@@ -5,9 +5,9 @@ SELECT
 FROM (
   -- Combine Aggregated Outputs
   SELECT
-    COALESCE(subq_27.visit__referrer_id, subq_38.visit__referrer_id) AS visit__referrer_id
-    , MAX(subq_27.visits) AS visits
-    , MAX(subq_38.buys) AS buys
+    COALESCE(subq_31.visit__referrer_id, subq_42.visit__referrer_id) AS visit__referrer_id
```

Comment:
confused how we ended up with MORE aliases after adding more optimization, which I would expect to reduce aliases 🤔 but alas, not particularly important!

Reply:
So it's not aliases, it's subgraph ID numbers. The reason for the change has to do with the shift in the number of calls to the `DataflowPlanBuilder.build_plan()` methods.

Basically, we add subquery aliases inside the DataflowToSqlQueryPlanConverter. This happens in like 20 different places in there, because it makes a lot of subqueries. Fair enough.

The thing is, we have another class - the DataflowPlanNodeOutputDatasetResolver - which we use both for creating source nodes and also for doing certain operations inside the DataflowPlanBuilder. This class is a direct subclass of the DataflowToSqlQueryPlanConverter, and when it does its work it basically creates a whole bunch of new subqueries, which means every time we add calls to it we change these subquery numbers.

That's also why we have these huge ID number values - we've basically gone over the input DAG a whole bunch of times, incrementing ID numbers all the way, before we produce the final query output. So an unoptimized query still has high numbers and a bunch of hard to explain gaps in the sequence.

Adding more optimizers to the set won't change this going forward, since we will still be generating the same number of plans (although the plan emitted by the optimizer might have totally different ID number ranges, because optimizers actually have their own ), but the update in ID numbers is one of the reasons why I didn't add the branch combiner to this PR.